frontend-service/src/main/java/com/amazonaws/blox/frontend/operations/EnvironmentController.java

```diff
@@ -24,27 +25,19 @@
 @Api
 @RequestMapping(path = "/v1/{cluster}/environments", produces = "application/json")
 public abstract class EnvironmentController {
-
   // Spring will autowire the current request into this field, so that we can get a handle on the
   // request context from API gateway.
   @Autowired HttpServletRequest request;
 
-  /** Get the caller's AWS Account ID from the current request. */
-  protected String getAwsAccountId() {
-    return getApiGatewayRequestContext().getAccountId();
-  }
+  @Autowired DataService dataService;
 
   /**
    * Get the API gateway request context from the current request.
    *
    * @return
    */
-  private ApiGatewayRequestContext getApiGatewayRequestContext() {
+  protected ApiGatewayRequestContext getApiGatewayRequestContext() {
```

Comment:
why protected?

Reply:
It's used by subclasses of EnvironmentController to get at the API gateway request context.